llvm/lib/Support/Triple.cpp

```diff
@@ -1087,12 +1087,11 @@ std::string Triple::normalize(StringRef Str) {
   }
 
   if (Arch == fpga64) {
-    Components.resize(4);
+    Components.resize(std::max(Components.size(), (size_t)3));
     if (Vendor == UnknownVendor)
       Components[1] = "xilinx";
-    if (Environment == UnknownEnvironment)
-      Components[3] = "sycldevice";
-    Components[2] = "unknown";
+    if (OS == UnknownOS)
+      Components[2] = "linux";
```

Comment:
What happens if this is not Linux?

Reply:
This will only add linux as a default if no OS is specified.